browser/search_engines/brave_template_url_service_factory_android.cc

```diff
@@ -0,0 +1,21 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/android/java/org/chromium/chrome/browser/search_engines/jni_headers/BraveTemplateUrlServiceFactory_jni.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_android.h"
+#include "chrome/browser/search_engines/template_url_service_factory.h"
+#include "components/search_engines/template_url_service.h"
+
+static TemplateURLService* GetTemplateUrlService(Profile* profile) {
+  return TemplateURLServiceFactory::GetForProfile(profile);
+}
+
+static base::android::ScopedJavaLocalRef<jobject>
+    JNI_BraveTemplateUrlServiceFactory_GetTemplateUrlService(JNIEnv* env, const
+    base::android::JavaParamRef<jobject>& j_profile) {
+  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile);
+  return GetTemplateUrlService(profile)->GetJavaObject();
```

Comment:
if this causes to use `browser/search_engines/private_window_search_engine_provider_service.cc` then it's awesome :+1:

Reply:
I actually asked Simon about this, and it is a mimic of `private_window_search_engine_provider_service`. That class itself is only applicable to extensions though, maybe useful in the future.